pallets/xtransfer/src/assets_wrapper.rs

```diff
@@ -151,24 +247,79 @@ pub mod pallet {
 			if let Some(asset) = IdToAsset::<T>::get(&asset_id) {
 				IdToAsset::<T>::remove(&asset_id);
 				AssetToId::<T>::remove(&asset);
+				// Unbind resource id and asset if have chain bridge set
+				if let Some(info) = IdToRegistryInfo::<T>::get(&asset_id) {
+					for bridge in info.enabled_bridges {
+						match bridge {
+							XBridge::ChainBridge {
+								chain_id: _,
+								resource_id,
+							} => {
+								ResourceIdToAssets::<T>::remove(&resource_id);
+							}
+							_ => {
+								continue;
+							}
+						}
+					}
+				}
+				// Delete registry info
+				IdToRegistryInfo::<T>::remove(&asset_id);
+
 				Self::deposit_event(Event::AssetUnRegistered { asset_id, asset });
 			}
 			Ok(())
 		}
+
+		#[pallet::weight(195_000_000)]
+		pub fn force_setup_chainbridge(
+			origin: OriginFor<T>,
+			asset_id: T::AssetId,
+			chain_id: u8,
+		) -> DispatchResult {
+			T::AssetsCommitteeOrigin::ensure_origin(origin)?;
+			if let Some(asset) = IdToAsset::<T>::get(&asset_id) {
+				let rid: [u8; 32] = asset.clone().into_rid(chain_id);
+				ensure!(
+					ResourceIdToAssets::<T>::get(&rid) == None,
+					Error::<T>::SolochainAlreadySetted
+				);
+				ResourceIdToAssets::<T>::insert(&rid, &asset);
+				// Save into registry info, here save chain id can not be added more than twice
+				if let Some(mut info) = IdToRegistryInfo::<T>::get(&asset_id) {
```

Comment:
If there's no registry info, I guess we should just throw an error?

Reply:
Error returned